cppcache/src/CacheImpl.hpp

```diff
@@ -140,7 +142,7 @@ class _GEODE_EXPORT CacheImpl : private NonCopyable, private NonAssignable {
    * Returns the distributed system that this cache was
    * {@link CacheFactory::create created} with.
    */
-  DistributedSystem& getDistributedSystem() const;
+  DistributedSystem& getDistributedSystem();
```

Comment:
Why drop the `const` here?

Reply:
The Cache class calls connect() and disconnect() on the returned value from getDistributedSystem(), which are non-const methods.  We put the const back in, and referred to the cacheImpl distributedSystem directly - let us know which way you prefer, or what option 3 might be.

Reply:
Nothing should be calling DS connection/disconnect directly anymore.